nbt/src/main/java/net/kyori/adventure/nbt/TagStringReader.java

```diff
@@ -38,6 +39,11 @@
 
   public CompoundBinaryTag compound() throws StringTagParseException {
     this.buffer.expect(Tokens.COMPOUND_BEGIN);
+    if(this.buffer.peek() == Tokens.COMPOUND_END) {
+      this.buffer.take();
+      return new CompoundBinaryTagImpl(new HashMap<>());
```

Comment:
```suggestion
      return CompoundBinaryTag.empty();
```

Reply:
Oh yeah I didn't see that the EMPTY map is wrapped into a new one when the tag is edited, I assumed it would make it immutable